File: library/src/test/java/me/proxer/library/api/messenger/MessagesEndpointTest.java
package me.proxer.library.api.messenger;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import me.proxer.library.entity.messenger.Message;
import me.proxer.library.enums.Device;
import me.proxer.library.enums.MessageAction;
import okhttp3.mockwebserver.MockResponse;
import org.assertj.core.api.Assertions;
import org.junit.Test;

import java.io.IOException;
import java.util.Date;
import java.util.List;

import static org.assertj.core.api.Java6Assertions.assertThat;

/**
 * @author Ruben Gees
 */
public class MessagesEndpointTest extends ProxerTest {

    @Test
    public void testDefault() throws ProxerException, IOException {
        server.enqueue(new MockResponse().setBody(fromResource("messages.json")));

        final List<Message> result = api.messenger()
                .messages()
                .build()
                .execute();

        assertThat(result).first().isEqualTo(buildTestMessage());
    }

    @Test
    public void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("messages.json")));

        api.messenger().messages()
                .conferenceId("123")
                .messageId("321")
                .markAsRead(false)
                .build()
                .execute();

        Assertions.assertThat(server.takeRequest().getPath())
                .isEqualTo("/api/v1/messenger/messages?conference_id=123&message_id=321&read=false");
    }

    private Message buildTestMessage() {
        return new Message("5193325", "131029", "121658", "RubyGee", "RubyGee",
                MessageAction.ADD_USER, new Date(1480260735L * 1000), Device.UNSPECIFIED);
    }
}

File: library/src/test/kotlin/me/proxer/library/api/messenger/MessagesEndpointTest.kt
package me.proxer.library.api.messenger

import me.proxer.library.ProxerTest
import me.proxer.library.entity.messenger.Message
import me.proxer.library.enums.Device
import me.proxer.library.enums.MessageAction
import me.proxer.library.runRequest
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test
import java.util.Date

/**
 * @author Ruben Gees
 */
class MessagesEndpointTest : ProxerTest() {

    private val expectedMessage = Message(
        id = "5193325", conferenceId = "131029", userId = "121658", username = "RubyGee", message = "RubyGee",
        action = MessageAction.ADD_USER, date = Date(1480260735L * 1000), device = Device.UNSPECIFIED
    )

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("messages.json") {
            api.messenger
                .messages()
                .build()
                .safeExecute()
        }

        result.first() shouldBeEqualTo expectedMessage
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("messages.json") {
            api.messenger.messages()
                .conferenceId("123")
                .messageId("321")
                .markAsRead(false)
                .build()
                .execute()
        }

        request.path shouldBeEqualTo "/api/v1/messenger/messages?conference_id=123&message_id=321&read=false"
    }
}
